doc/modularity.rst

```diff
@@ -62,7 +62,8 @@ Without modules, packages with the highest version are used by default.
 Module streams can distribute packages with lower versions than available in the
 repositories available to the operating system. To make such packages available
 for installs and upgrades, the non-modular packages are filtered out when they match
-by name with modular packages from any existing stream.
+by name or provide with modular package names from any existing stream. Modular source packages
+do not cause filtering out binary packages.
```

Comment:
How about this formulation to clean it up a bit?

> ... the non-modular packages are filtered out when their name or provide matches against a modular package name from any existing stream. Modular source packages will not cause non-modular binary packages to be filtered out.

Reply:
@lukash Good suggestion, thanks. I was still not happy about `existing stream` therefore I replaced it.